component.ts

```diff
@@ -0,0 +1,47 @@
+function fetchResults(query: string) {
```

Comment:
```suggestion
import React, {useState, useEffect} from "react"
function fetchResults(query: string) {
```

Même si on n'utilise pas React directement, ça doit être importe pour utiliser du JSX
D'ailleurs il faudrait que l'extension de ce fichier soit `.tsx` pour que ça marche d'ailleurs !